pkg/reconciler/route/reconcile_resources.go

```diff
@@ -123,37 +125,101 @@ func (c *Reconciler) reconcileClusterIngress(
 	return clusterIngress, err
 }
 
-func (c *Reconciler) reconcilePlaceholderService(ctx context.Context, route *v1alpha1.Route, ingress *netv1alpha1.ClusterIngress) error {
+func (c *Reconciler) deleteServices(namespace string, svcs map[string]*corev1.Service) error {
+	for _, svc := range svcs {
+		if err := c.KubeClientSet.CoreV1().Services(namespace).Delete(svc.Name, nil); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *Reconciler) getServicesAsMap(route *v1alpha1.Route) (map[string]*corev1.Service, error) {
```

Comment:
the code using this seems not to use the values. So maybe k8s stringset is more suitable here? E.g. in `deleteServices` above, you just use `svc.Name`, which is the _key_ to the map in the first place.

Reply:
Neat, didn't know they had a `sets` package.